addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1387,11 +1387,8 @@ export class OdooEditor extends EventTarget {
                     link &&
                     !isBlock(link) &&
                     link.textContent !== '' &&
-                    !(
-                        // Ignore links wrapped around a single image.
-                        link.children.length === 1 &&
-                        link.firstElementChild.nodeName === 'IMG'
-                    )
+                    // Only add the ZWS for simple (possibly styled) text links.
+                    ![...link.children].some(isBlock)
```

Comment:
This still leads to: 
![image](https://github.com/odoo/odoo/assets/10338094/4fb7cacf-9904-4d92-a7ea-e692cd2a62cb)

The structure is basically `<a><span><img/></span></a>` (the span coming from the t-field implementation). So here the only child (the span) is inline, not block. But actually, it would be the same thing with `<a><img/></a>` as images are inline by default.

Note also: I wondered why the above line `link.textContent !== ''` was not enough to handle the logo case actually: that's because of whitespaces. Maybe that line should be reviewed/removed as well?

I am not sure what the solution should be. Assuming we need to keep adding zero-width-spaces (I understand the need, but maybe there is another way?), maybe a simpler thing to do would:
- Always adding them in links (no condition here)
- Setting them as `position: fixed/absolute;` -> that way they should not participate in the link inner content flow. (To test)

Reply:
Weird, when I test I don't have that, I have this instead:
![image](https://github.com/odoo/odoo/assets/24205914/4b5fa009-a134-4c25-a9b8-9c112894d3ac)

What am I doing differently?

> Note also: I wondered why the above line link.textContent !== '' was not enough to handle the logo case actually: that's because of whitespaces. Maybe that line should be reviewed/removed as well?

Good catch, `link.textContent !== ''` should probably be replaced by `link.textContent.replace(/^\s*|\s*$/g, '') !== ''`. I'll look into it.

> maybe there is another way

This is our so-manieth attempt at handling this issue with a different approach and it's always incredibly tricky. So there are other ways but so far the ones we tried were unfortunately less convincing than this one. We'll try and keep refining this one for the moment... I like your idea of `position: fixed/absolute`, if it doesn't throw off the selection it would be a great solution!

Reply:
> What am I doing differently?

I tested with 17.0 (cherry-picking your commit), and on Firefox (in local, with only website installed)

Reply:
@qsm-odoo done. The condition was indeed wrong.